pkg/types/azure/machinepool.go

```diff
@@ -50,6 +52,13 @@ type MachinePool struct {
 	// +optional
 	VMNetworkingType string `json:"vmNetworkingType,omitempty"`
 
+	// BootDiagnostics has the value for the storage account URI where the
+	// machine log information is sent to. Defaults to disabled if no value
+	// and defaults to the installer created storage account if the URI is not
+	// mentioned.
```

Comment:
These are sensible defaults (they preserve current behavior), but I would suggest changes that imho offer some advantages & simplify things for us:

- Default to Managed. The managed storage account is cheap. I'm not aware of any reasons not to do this.
- Don't default to the installer-created storage account. The managed storage account serves the same purpose and we should really destroy the installer-created storage account. This will simplify your code as you were needing to jump through some hoops for the storage account name.

The UserManaged StorageAccountURI should have some dynamic validation to check that the user provided account really exists to prevent issues during infra provisioning. 

As @jhixson74 mentioned we also need to take into account compute nodes. On that note, we should check the ipi-install-install ci step as it is setting bootdiagnostics in the manifests.